algorithms/indexing/indexer.py

```diff
@@ -654,12 +654,12 @@ def index(self):
                         # with this deleted experiment - indexed flag removed
                         # below
                         last = len(experiments)
-                        sel = refined_reflections["id"] == last
+                        sel = reflections_for_refinement["id"] == last
```

Comment:
The original code was correct. This line is only hit in multi-lattice indexing, and there was a refinement failure after indexing a second or subsequent lattice - so the `refined_reflections` variable exists from the previous loop iteration. In which case, we delete the last experiment and remove the experiment ids associated with this experiment from the reflection table and reset the ids.

Reply:
See also https://github.com/dials/dials/commit/8f30e07556ec99216f03668e9cc0de857e5a8c66

Reply:
Ah, OK, sorry, I just couldn't work it out from the code.  I'll revert that commit.